### PR TITLE
(#21) Test : User is able to filter based on location in the CTS advanced search

### LINKED ISSUE #21: Story: User is able to filter based on location in the CTS advanced search
## As a user, I want to enter location criteria, so that I can filter CTS results based on Location

> **ESTIMATE** 20

### Acceptance Criteria

```gherkin
Scenario: User is able to filter results using the Veteran Affairs switch
Given user is navigating to "/about-cancer/treatment/clinical-trials/search/advanced"
When user clicks on "Limit results to Veterans Affairs facilities" switch
Then the following radio buttons are displayed
|radioButton | visibility|
|Search All Locations| displayed|
|ZIP Code| displayed|
|Country, State, City| displayed|
|Hospitals/Institutions| none|
|At NIH (only show trials at the NIH Clinical Center in Bethesda, MD)|none|

Scenario: User is able to filter results using the Veteran Affairs switch 
  Given user is navigating to "/about-cancer/treatment/clinical-trials/search/advanced"
    When  user clicks on "Limit results to Veterans Affairs facilities" switch
    And user clicks "Find Trials" button
    Then the search is executed and results page is displayed
    And the url query has the following corresponding code
    |queryParam|value|
    |loc|0|
    |va|1|


Scenario: User is able to filter results by ZipCode using the Veteran Affairs switch 
  Given user is navigating to "/about-cancer/treatment/clinical-trials/search/advanced"
    When  user selects "ZIP Code" radio button
    And user types "20105" in the ZIP Code field 
    And user selects "50 miles" in the Radius field
    And user clicks "Find Trials" button
    Then the search is executed and results page is displayed
    And the url query has the following corresponding code
    |queryParam|value|
    |loc|1|
    |rl|2|
    |z|20105|
    |zp|50|

Scenario: User is able to filter results by Country,State,City
  Given user is navigating to "/about-cancer/treatment/clinical-trials/search/advanced"
    When  user selects "Country,State,City" radio button
    And user selects "United States" in the Country field 
    And user types "Virginia" in the State field
    And user selects matched State from autosuggest
    And user types "Arlington" in the City field
    And user clicks "Find Trials" button
    Then the search is executed and results page is displayed
    And the url query has the following corresponding code
    |queryParam|value|
    |loc|2|
    |rl|2|
    |lcnty|United States|
    |lcty|Arlington|

Scenario: User is able to filter results by Hospitals/Institutions
  Given user is navigating to "/about-cancer/treatment/clinical-trials/search/advanced"
    When  user selects "Hospitals/Institutions" radio button
    And user types "Massachusetts General Hospital" in the Hospitals/Institutions field
    And user selects matched Hospitals/Institutions from autosuggest
    And user clicks "Find Trials" button
    Then the search is executed and results page is displayed
    And the url query has the following corresponding code
    |queryParam|value|
    |loc|3|
    |rl|2|
    |hos|Massachusetts General Hospital|

Scenario: User is able to filter results by atNIH
  Given user is navigating to "/about-cancer/treatment/clinical-trials/search/advanced"
    When  user selects "At NIH" radio button
    And user clicks "Find Trials" button
    Then the search is executed and results page is displayed
    And the url query has the following corresponding code
    |queryParam|value|
    |loc|4|
    |rl|2|
   

    
```

### Resources:
* Feature: JIRA Ticket #
* Mockups: *Here goes a URL to or the name of the mockup(s) in inVision*
* Web Page: *Here goes a URL to an existing web page*

### Notes

*Some complementary notes if necessary:*

* > Here goes a quote from an email
* Here goes whatever useful information can exist…

---

### Prerequisites
- *required dependency for identifying the problem, scope and then solution for this issue*
- [ ] Short Spike Description - Issue #9999

### Tasks
- [ ] Short Task Description - Issue #9999
- [ ] Short Task Description - Issue #9999

### Technical Notes
- More notes, or technical nuances/direction

### PR DESCRIPTION
**Pull request by SupriyaSaxena0605**
closes #21

### CODE REVIEW

File: cypress/integration/CTS/LocationSection/LocationSection/locationSectionTest.js [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+And("user selects {string} in the Country field", (country) => {
+    cy.get('select#country').select(country);
+});
+
+And("user types {string} in the State field", (state) => {
+    cy.get('input#lst').type(state);
+});
+
+And("user selects matched State from autosuggest", () => {
+    cy.get('.cts-autocomplete__menu-item.highlighted').click();
```
Comment (kate-mashkina) [inline cypress/integration/CTS/LocationSection/LocationSection/locationSectionTest.js:87]: This step and the "user selects matched Hospital from autosuggest" seems to be doing exactly same thing in the code base.
Why can't you just combine them into one step "user selects matched item from autosuggest"?

File: cypress/integration/CTS/LocationSection/LocationSection/locationSectionTest.js [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+});
+
+And("user types {string} in the City field", (city) => {
+    cy.get('input#city').type(city);
+});
+
+And("user types {string} in the Hospital field", (hospital) => {
+    cy.get('.search-location__block').scrollIntoView();
+    cy.get('fieldset#fieldset--location > .cts-fieldset__body .cts-autocomplete  input[role^="combobox"]').type(hospital);
```
Comment (kate-mashkina) [inline cypress/integration/CTS/LocationSection/LocationSection/locationSectionTest.js:96]: locator is long and appears to be fragile - could you replace it with something shorter, like `div.search-location__block>div`

File: cypress/integration/CTS/LocationSection/LocationSection/locationSectionTest.js (hunk 1 above)
Comment (SupriyaSaxena0605) [reply cypress/integration/CTS/LocationSection/LocationSection/locationSectionTest.js:87]: fixed it

File: cypress/fixtures/example.json [hunk 3]
```diff
@@ -0,0 +1,5 @@
+{
+  "name": "Using fixtures to represent data",
+  "email": "hello@cypress.io",
+  "body": "Fixtures are a great way to mock data for responses to routes"
+}
```
[the remark this answers is not in the document]
Comment (SupriyaSaxena0605) [reply cypress/fixtures/example.json]: deleted it

File: cypress/integration/CTS/LocationSection/LocationSection.feature [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+        And user clicks "Find Trials" button
+        Then the search is executed and results page is displayed
+        And the url query has the following corresponding code
+            | queryParam | value |
+            | loc        | 0     |
+            | rl         | 2     |
+            | va         | 1     |
+
+
+    Scenario: User is able to filter results by ZipCode using the Veteran Affairs switch
```
[the remark this answers is not in the document]
Comment (SupriyaSaxena0605) [reply cypress/integration/CTS/LocationSection/LocationSection.feature]: fixed it

File: cypress/integration/CTS/LocationSection/LocationSection/locationSectionTest.js (hunk 2 above)
Comment (SupriyaSaxena0605) [reply cypress/integration/CTS/LocationSection/LocationSection/locationSectionTest.js]: fixed it

**Pull request merged**